x/cosmwasmpool/model/pool.go

```diff
@@ -83,7 +83,13 @@ func (p Pool) SpotPrice(ctx sdk.Context, quoteAssetDenom string, baseAssetDenom
 	if err != nil {
 		return osmomath.BigDec{}, err
 	}
-	return osmomath.MustNewBigDecFromStr(response.SpotPrice), nil
+
+	result, err := osmomath.NewBigDecFromStr(response.SpotPrice)
+	if err != nil {
+		return osmomath.BigDec{}, err
+	}
+
+	return result, nil
```

Comment:
Note: This is a precaution and a follow-up to a review discussion.

Ref: https://github.com/osmosis-labs/osmosis/issues/6406